### DISCUSSION TITLE
props.route现在还有地方可以取吗

### DISCUSSION THREAD

**Discussion by Ruboyq** [umijs/umi#8973] [Q&A] [answered]
之前的umi3工程是自己写的ProLayout，然后route是通过props.route传进去的，现在想搞umi4，我看文档都是写的layout插件，搞运行时配置，但是老代码的布局是写在BasicLayout里的，还是按图里这样写没有菜单了，请问应该怎么修改呢
![image](https://user-images.githubusercontent.com/26861176/184606170-31641cca-607e-4118-ac7f-dff425f6ae25.png)

**Comment by fz6m** [comment, marked as answer]
https://umijs.org/docs/api/api#useroutedata

**Reply by Ruboyq** [reply]
这个我试过，拿到的是当前组件的路由信息，缺少子路由，我这里是需要整个routes塞给prolayout来渲染左边的菜单，个人理解应该是umi处理过的config.ts里的routes
![image](https://user-images.githubusercontent.com/26861176/184626184-25fbd35c-7f14-40fc-a891-01e9013cf62d.png)
（useRouteData）
![image](https://user-images.githubusercontent.com/26861176/184626280-2b901b05-b717-436e-9b25-edc017a367ca.png)
（没有塞route，菜单无了）

**Reply by fz6m** [reply]
```ts
const appData = useAppData()
const { clientRoutes } = appData
```

**Comment by Ruboyq** [comment]
感谢回复，我解析了一下就好了，不过icon需要自己转换成<Icon>，clientRoutes里的icon是字符串
![image](https://user-images.githubusercontent.com/26861176/184630447-146f2b3a-e79d-4f96-a88c-6aae55e76c8f.png)
